modules/core/src/main/java/org/apache/ignite/spi/deployment/local/LocalDeploymentSpi.java

```diff
@@ -146,6 +150,68 @@
         return null;
     }
 
+    /** {@inheritDoc} */
+    @Nullable @Override public DeploymentResource findResource(String rsrcName, @Nullable ClassLoader clsLdr) {
+        assert rsrcName != null;
+
+        if (clsLdr != null) {
+            ConcurrentMap<String, String> rsrcs = ldrRsrcs.get(clsLdr);
+
+            if (rsrcs == null)
+                return null;
+
+            return findResource0(rsrcs, rsrcName, clsLdr);
+        }
+
+        // we can remove this stub after deprecated IgniteCompute.localDeployTask was deleted.
+        for (Entry<ClassLoader, ConcurrentMap<String, String>> e : ldrRsrcs.descendingEntrySet()) {
+            ClassLoader ldr = e.getKey();
+            ConcurrentMap<String, String> rsrcs = e.getValue();
+
+            DeploymentResourceAdapter res = findResource0(rsrcs, rsrcName, ldr);
+
+            if (res != null)
+                return res;
+        }
+
+        return null;
+    }
+
+    /**
+     * Finds appropriate resource.
+     *
+     * @param rsrcs Resources.
+     * @param rsrcName Class name or class alias to find class loader for.
+     * @param clsLdr desired class loader.
+     * @return Deployed class loader, or {@code null} if not deployed.
+     */
+    @Nullable private DeploymentResourceAdapter findResource0(Map<String, String> rsrcs, String rsrcName, ClassLoader clsLdr) {
+        String clsName = rsrcs.get(rsrcName);
+
+        // Return class if it was found in resources map.
+        if (clsName != null) {
+            // Recalculate resource name in case if access is performed by
+            // class name and not the resource name.
+            rsrcName = getResourceName(clsName, rsrcs);
+
+            assert clsName != null;
+
+            try {
+                Class<?> cls = U.forName(clsName, clsLdr);
+
+                assert cls != null;
+
+                // Return resource.
+                return new DeploymentResourceAdapter(rsrcName, cls, clsLdr);
+            }
+            catch (ClassNotFoundException ignored) {
```

Comment:
Print a warning, otherwise we never find a problem.

Reply:
done